server/src/main/java/org/elasticsearch/action/admin/cluster/settings/TransportClusterUpdateSettingsAction.java

```diff
@@ -185,17 +185,6 @@ public void onAckTimeout() {
             }
 
             private void reroute(final boolean updateSettingsAcked) {
-                // We're about to send a second update task, so we need to check if we're still the elected master
```

Comment:
I'm not very experienced with the code here but if it's as simple as that the reroute task will return NotMasterException if the master changed in the meantime, it sounds good to me. I wonder @DaveCTurner if a test is possible (to ensure that it's handled correctly if the master changed in the meantime) or at least a comment to say that the following reroute task may fail with NotMasterException and it's fine.

Reply:
(I think you might be right btw, it looks like the `BatchedRerouteService` in fact does not return a publish-failure exception on a publication failure)

Reply:
Ok I fixed that in #97224.

I'm not going to add tests in this PR. I have sketched out adding batching support to this action, which I'll open as a draft PR shortly to remind us to clean up this debt. If I added tests here too then they'd need to be reworked to handle the batching changes.

Reply:
See https://github.com/elastic/elasticsearch/pull/97246 for the batching PR